CodePush.js

```diff
@@ -436,6 +477,10 @@ if (NativeCodePush) {
       DOWNLOADING_PACKAGE: 7,
       INSTALLING_UPDATE: 8
     },
+    SyncMode: {
+      ON_APP_START: 0,
+      ON_APP_RESUME: 1,
```

Comment:
This is a random thought, I'd be interested in hearing what you think. Would it be worth adding a `MANUAL` option here? This would basically just result in `notifyApplicationReady()` being injected into `componentDidMount()` and nothing else (for now). Our docs could just document the `MANUAL` option as an alternative way of using CodePush, where you call `sync()` wherever you need it (doesn't matter if it's on app start or not). Then `sync()` is not technically an alternative to `codePushify()`, it's just something you can use in `MANUAL` mode.

For example, if someone wants to do a codePush update on button press (who knows why), then they wouldn't need to know about `notifyApplicationReady()`, they can just call `sync()` on the button press and it works as they expect.